### PR TITLE
add take/free instance(s) API

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#1896

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This change adds dummy APIs for take/free instance and the batch mode.

### Tests

- [X] The following tests are written for this issue:

NA

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class MaintenanceManagementInstanceInfo {
+
+  public enum Status {
```
Comment (junkaixue) [inline helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java:28]: Let's make it more detailed naming? OperationalStatus?

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    SUCCESS,
+    FAILURE
+  }
+  private List<Object> operationResult;
```
Comment (junkaixue) [inline helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java:32]: Shall we make it as a map for searching purpose? Or just one String. Not quite sure about the list of string.

File: helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.apache.helix.rest.common.datamodel;
+
+
+public class RestSnapShot {
```
Comment (junkaixue) [inline helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java:4]: Have a kv map there at least? And simple kv set/get methods.

Comment (junkaixue) [reply helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java:4]: Add a comment at class level that this Snapshot can extend Snapshot from common/core module if there is more generic snapshot.

Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java:4]: Thanks for the review. I plan to add the enum and other implementation for this class in further PR.

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java (hunk 1 above)
Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java]: TFTR. Updated.

File: helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java (hunk 3 above)
Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/common/datamodel/RestSnapShot.java:25]: Add a note for that. Will do the impl in next PR.

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class MaintenanceManagementInstanceInfo {
+
+  public enum OperationalStatus {
+    SUCCESS, FAILURE
```
Comment (junkaixue) [inline helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java:29]: NIT: let's format it as line by line.

**Review by junkaixue — APPROVED** [review]
Overall, lgtm

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java (hunk 4 above)
Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementInstanceInfo.java]: TFTR. Updated.

**Comment by xyuanlu** [comment]
This PR is ready to be merged. Approved by @junkaixue . 
Commit message: "Create Cluster Maintenance Management service and add API signature and interfaces."

**Pull request merged**